src/commands/quote.ts

```diff
@@ -0,0 +1,105 @@
+import { CommandInteraction } from 'discord.js'
+import { Discord, SimpleCommand, SimpleCommandMessage, SimpleCommandOption, Slash, SlashOption } from 'discordx'
+import fetch from 'node-fetch'
+
+interface Quote {
+  id: number
+  body: string
+}
+
+interface QuoteData {
+  error: boolean
+  type: number
+  total: number
+  data: Array<Quote>
+}
+
+@Discord()
+class quoteCommand {
+  private endpoint = 'https://api.bufutda.com/bot/quote?channel=bananasaurus_rex'
+  private quotes: Array<Quote> = []
+  private lastFetch: Date = new Date()
+  private cacheDuration = 1 * 60 * 1000 // 30 minutes in milliseconds
+
+  @Slash('quote', { description: 'Get server quote' })
+  private async quoteSlash(
+    @SlashOption('quoteid', { type: 'NUMBER', required: false })
+    id: number,
+    interaction: CommandInteraction
+  ) {
+    await interaction.deferReply()
+
+    await this.updateQuotes().then(async () => {
+      if (id) {
+        const quote = this.quotes.find((quote) => quote.id == id)
+        if (!quote) {
+          await interaction.followUp(`Unable to find quote with ID ${id}`)
+        } else {
+          await interaction.followUp(quote.body)
+        }
+      } else {
+        const randomQuote = this.getRandomQuote()
+        if (randomQuote) {
+          await interaction.followUp(randomQuote.body)
+        } else {
+          await interaction.followUp('Unable to get quotes.')
+        }
+      }
+    })
+  }
+
+  @SimpleCommand('quote', { description: 'Get server quote', argSplitter: '\n' })
+  private async quoteSimple(
+    @SimpleCommandOption('id', { type: 'NUMBER' })
+    id: number,
+    command: SimpleCommandMessage
+  ) {
+    await this.updateQuotes().then(async () => {
+      const channel = command.message.channel
+      if (id) {
+        const quote = this.quotes.find((quote) => quote.id == id)
+        if (!quote) {
+          await channel.send(`Unable to find quote with ID ${id}`)
+        } else {
+          await channel.send(quote.body)
+        }
+      } else {
+        const randomQuote = this.getRandomQuote()
+        if (randomQuote) {
+          await channel.send(randomQuote.body)
+        } else {
+          await channel.send('Unable to get quotes.')
+        }
+      }
+    })
+  }
+
+  private getRandomQuote(): Quote {
+    return this.quotes[Math.floor(Math.random() * this.quotes.length)]
```

Comment:
This could throw an error if the quotes array is empty that could crash the bot... I'll make a fix in main.js to catch errors on command calls to prevent them. Until then this function should make sure the list isn't empty otherwise throw undefined or some dummy quote.

Reply:
I'm checking for null now.